test-suite/asianoptions.cpp

```diff
@@ -692,15 +755,27 @@ void AsianOptionTest::testDiscreteGeometricAveragePriceHestonPastFixings() {
 
     ext::shared_ptr<PricingEngine> mcEngine =
         MakeMCDiscreteGeometricAPHestonEngine<LowDiscrepancy>(hestonProcess)
-        .withSamples(32767)
+        .withSamples(10000)
         .withSeed(43);
 
     Option::Type type(Option::Call);
     Average::Type averageType = Average::Geometric;
 
+    auto strike_index = -1;
```

Comment:
Instead of adding an external index and keep track of it, I'd change the loops to `(for Size i=0; i<strikes.size(); ++i)` and similar and use `strikes[i]` in the loop.

However, I wouldn't be opposed to set the tolerance to 0.06 for all cases and be done with it.

Reply:
I'll change it to for() as you suggested, will keep the tolerance tight if that's ok.